pkg/store/consul/transaction/transaction.go

```diff
@@ -35,10 +35,20 @@ type tx struct {
 	committedMu sync.Mutex
 }
 
+// New() returns a new context derived from the one passed as an argument and
+// its cancel function with the context containing metadata used to build a
+// consul transaction. If the passed context already has consul operations
+// defined, the new context will inherit those operations defined but in a
+// separate transaction.
 func New(ctx context.Context) (context.Context, context.CancelFunc) {
```

Comment:
ah, I guess it's not so different than what we have today - I forgot about the context.Background thing.

I'm ambivalent about this. I don't see there being a use case for the "multi-method" approach for transaction.New() but the API seems fine as it is.